android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java

```diff
@@ -139,13 +154,15 @@ public void addAndroidListener(String eventName) {
     @ReactMethod
     public void removeAndroidListeners(int count) {
         Logger.info("UrbanAirshipReactModule - Event listeners removed: " + count);
-        listenerCount -= count;
-        if (listenerCount < 0) {
-            listenerCount = 0;
+
+        EventEmitter.shared().listenerCount -= count;
+        if (EventEmitter.shared().listenerCount < 0) {
```

Comment:
not a critical thing, but you could also write this as:

```
long currentCount = EventEmitter.shared().listenerCount;
EventEmitter.shared().listenerCount = max(0, currentCount - count);
```

Or something. Sometimes it can be easier to think in terms of clipping value ranges, and if it were behind a getter/setter you could encapsulate the clipping there. Up to you though.